packages/next/next-server/lib/router/router.ts

```diff
@@ -489,6 +489,14 @@ export default class Router implements BaseRouter {
           }
 
           Router.events.emit('beforeHistoryChange', as)
+
+          const { url: curUrl, as: curAs, options: curOptions } = history.state
+
+          this.changeState('replaceState', curUrl, curAs, {
+            ...curOptions,
+            scrollY: window.scrollY,
```

Comment:
We should namespace this for Next.js probably.

`_N_Y` and `_N_X` or something.

Reply:
Good point, updated